src/core/attribute.d

```diff
@@ -55,3 +55,60 @@ version (D_ObjectiveC) struct selector
 {
     string selector;
 }
+
+/**
+ * Use this attribute to declare an ABI tag on a C++ symbol.
+ *
+ * ABI tag is an attribute introduced by the GNU C++ compiler.
+ * It modifies the mangled name of the symbol to incorporate the tag name,
+ * in order to distinguish from an earlier version with a different ABI.
+ * See: $(LINK2 https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Attributes.html, GCC attributes documentation).
+ *
+ * This is a special compiler recognized attribute, it has a few
+ * requirements, which all will be enforced by the compiler:
+ *
+ * $(UL
+ *  $(LI
+ *      The attribute can only be attached to a declaration of
+ *      an $(D_CODE extern(C++)) variable, function, aggregate,
+ *      enum, template, or namespace.
+ *  ),
+ *
+ *  $(LI
+ *      The string arguments must only contain valid characters
+ *      for C++ name mangling which currently include alphanumerics
+ *      and the underscore character.
+ *  ),
+ * )
+ *
+ * Note: Unlike other attributes, if applied to a namespace,
+ * this attribute it is not inherited by the members.
+ *
+ * Examples:
+ * ---
+ * @gnu_abi_tag("bar")
+ * extern (C++, Foo)
+ * class A
+ * {
+ *     @gnu_abi_tag("C")
+ *     void method() {}
+ * }
+ *
+ * @gnu_abi_tag("B")
+ * __gshared A var;
+ *
+ * @gnu_abi_tag("E")
+ * enum ABC { a, b, c }
+ * ---
+ */
+version (Posix) struct gnu_abi_tag
```

Comment:
Please prefer naming conventions documented in the [D Style](https://dlang.org/dstyle.html).